pkg/types/types.go

```diff
@@ -18,6 +16,6 @@ type Root = types.Root
 type ValidatorIndex = uint64
 
 type Coordinate struct {
-	Slot Slot
+	Slot primitives.Slot
```

Comment:
I had made this `types` module so that the actual implementations could be shielded from the rest of this code base in the event we wanted to swap them out

what do you think about restoring the use of the local `types.Slot` etc and then just changing the definitions here?